skorch/net.py

```diff
@@ -758,11 +758,24 @@ def initialize_optimizer(self, *args, **kwargs):
           pass the named parameters to :meth:`.get_params_for_optimizer`.
 
         """
+        # Note: we have to filter out potential duplicate parameters. This can
+        # happen when a module references another module (e.g. the criterion
+        # references the module), thus yielding that module's parameters again.
```

Comment:
This could be silly question: What is an example of a criterion with learnable parameters?

Reply:
Not silly. One use case that has been discussed at some point is for GAN training with the generator being the `module` and the discriminator the `criterion` (though you might need different criteria in this case). Another case (which revealed this bug for me) is when using GPyTorch.
But this issue is not only relevant when the criterion references the module, it could also be one module referencing a second module.